### PR TITLE
Add chapter about generating markdown to guide

### PR DESCRIPTION
**Pull request by TheNeikos**
@matthiasbeyer would you mind doing a quick proof read?

### CODE REVIEW

File: doc_guide/src/generating_markdown.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Generating Markdown
+
+Depending on your case you can interact with `TypeDescription`s in multiple ways.
+
+If you are given a JSON formatted type description file, then you should use the CLI for further processing.
```
Comment (matthiasbeyer) [inline doc_guide/src/generating_markdown.md:5]:
```suggestion
If you are given a JSON formatted type description file, then you can use the CLI for further processing.
```

File: doc_guide/src/generating_markdown.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+- With the nix package manager (flakes enabled): `nix run github:TheNeikos/type_description`
+- With cargo: `cargo install type_description`
+- Directly from source: `git clone https://github.com/TheNeikos/type_description && cd type_description && cargo run -F bin`
+
+Either way you have installed it. You can now generate markdown from it by sending it to standard input.
+
+E.g.:
+
+- Using nix: `cat desc.rtd | nix run github:TheNeikos/type_description`
```
Comment (matthiasbeyer) [inline doc_guide/src/generating_markdown.md:25]: What is `desc.rtd`?

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]

File: doc_guide/src/generating_markdown.md (hunk 2 above)
Comment (TheNeikos) [reply doc_guide/src/generating_markdown.md:25]: Haha, just a made up filename of where the 'rust type description' could be. I'll change it!

Comment (matthiasbeyer) [reply doc_guide/src/generating_markdown.md:25]: Yeah, it is somewhat surprising. I actually looked at the repository whether there's a file named `desc.rtd`. I also am not sure whether the `example.com` is a good idea... maybe replace it by something like `<your description file>` or `<your description URL>`!

**Comment by TheNeikos** [comment]
bors merge

**Comment by TheNeikos** [comment]
bors retry

**Pull request merged**